docs/src/main/sphinx/security/inspect-pem.rst

```diff
@@ -0,0 +1,139 @@
+
+    cat clustercoord.pem
+
+A key file looks something like the following:
+
+.. code-block:: text
+
+  -----BEGIN PRIVATE KEY-----
+  MIIEowIBAAKCAQEAwJL8CLeDFAHhZe3QOOF1vWt4Vuk9vyO38Y1y9SgBfB02b2jW
+  ....
+  -----END PRIVATE KEY-----
+
+If your key file reports ``BEGIN ENCRYPTED PRIVATE KEY`` instead, this means the
+key is encrypted and you must use the password to open or inspect the key. You
+may have specified the password when requesting the key, or the password could
+be assigned by your site's network managers.
+
+  If your key file reports ``BEGIN EC PRIVATE KEY`` or
+  ``BEGIN DSA PRIVATE KEY``, this designates keys using Elliptical Curve or
+  DSA alternatives to RSA. In these cases, consult ``openssl`` references for
+  the appropriate versions of the verification commands below.
+
+Inspect PEM certificate files
+-----------------------------
+
+A separate certificate file looks like the following example with the ``cat``
+command:
+
+.. code-block:: text
+
+  -----BEGIN CERTIFICATE-----
+  MIIDujCCAqICAQEwDQYJKoZIhvcNAQEFBQAwgaIxCzAJBgNVBAYTAlVTMRYwFAYD
+  ....
+  -----END CERTIFICATE-----
+  -----BEGIN CERTIFICATE-----
+  MIIDwjCCAqoCCQCxyqwZ9GK50jANBgkqhkiG9w0BAQsFADCBojELMAkGA1UEBhMC
+  ....
+  -----END CERTIFICATE-----
+
+The file can show a single certificate section, or more than one to express a
+chain of authorities, each certifying the previous.
+
+If you received a single file, make sure it shows at least one ``KEY`` and one
+``CERTIFICATE`` section. If you received separate files, concatenate them into
+one, typically in order from key to certificate. For example:
+
+.. code-block:: shell
+
+  cat clustercoord.key clustercoord.cert > clustercoord.pem
+
+.. _validate_pems:
+
+Validate PEM keys
+-----------------
+
+This page presumes your system provides the ``openssl`` command from OpenSSL 1.1
+or later.
+
+Test an RSA private key's validity with the following command:
+
+.. code-block:: text
+
+  openssl rsa -in clustercoord.pem -check -noout
+
+Look for the following confirmation message:
+
+.. code-block:: text
+
+  RSA key ok
+
+Validate PEM certificates
+-------------------------
+
+Analyze a certificate file, or the certificate portion of a concatenated file,
+with the following ``openssl`` command:
+
+.. code-block:: text
+
+  openssl x509 -in server.pem -text -noout
+
+If your certificate was generated with a password, ``openssl`` prompts for it.
+
+In the output of the ``openssl`` command, look for the following
+characteristics:
+
+*  Modern browsers now enforce 398 days as the maximum validity period for a
+   certificate. Look for ``Not Before`` and ``Not After`` dates in the
+   ``Validity`` section of the output, and make sure the time span does not
+   exceed 398 days.
+*  Modern browsers and clients require the **Subject Alternative Name** (SAN)
+   field. Make sure this shows the DNS name of your server, such as
+   ``DNS:clustercoord.example.com``. Certificates without SANs are not
```

Comment:
again .. maybe change to trino.example.com ?

Reply:
The clustercoord example name deliberately covers Trino and any upstream branches equally well without favorites.

Reply:
ok .. good point.